File: sdkwork-im-sdk-flutter/composed/lib/src/session_module.dart
import 'package:im_sdk_generated/im_sdk_generated.dart';

import 'context.dart';

class ImSessionModule {
  final ImSdkContext context;

  ImSessionModule(this.context);

  Future<SessionResumeView?> resume(ResumeSessionRequest body) {
    return context.transportClient.session.resume(body);
  }

  Future<PresenceSnapshotView?> disconnectDevice(PresenceDeviceRequest body) {
    return context.transportClient.session.disconnect(body);
  }
}

File: sdkwork-im-sdk-rust/composed/src/session_module.rs
use crate::ImSdkContext;
use sdkwork_im_sdk_generated::{
  PresenceDeviceRequest,
  PresenceSnapshotView,
  ResumeSessionRequest,
  SdkworkError,
  SessionResumeView,
};

#[derive(Clone)]
pub struct ImSessionModule {
  context: ImSdkContext,
}

impl ImSessionModule {
  pub(crate) fn new(context: ImSdkContext) -> Self {
    Self { context }
  }

  pub async fn resume(&self, body: ResumeSessionRequest) -> Result<SessionResumeView, SdkworkError> {
    self.context.transport_client().session().resume(&body).await
  }

  pub async fn disconnect(
    &self,
    body: PresenceDeviceRequest,
  ) -> Result<PresenceSnapshotView, SdkworkError> {
    self.context.transport_client().session().disconnect(&body).await
  }
}

